### PR TITLE
Prepare repository for next `.github` release

### PR DESCRIPTION
**Pull request by alejandrohdezma**
# What has been done in this PR?

## Prepare project for next `.github` release

- Add `ci-publish` command alias that will be used instead of `ci-release` for publishing artifacts.
- Move docs templates to `docs` folder (as mdoc is expected).

## Simplify SBT build

- Re-order settings.
- Remove unneeded settings.
- Inline settings in `build.sbt` (since it is not too big).

## Simplify dependencies

Both `munit` & `docker-testkit-core` dependencies have been set as `Provided` with a version range (instead of a fixed one). This will allow users of our
library to use different versions of both dependencies as long as they comply with our range. This eases mantainability since we don't need to keep releasing new versions of the library as long as the APIs that we use from the two libraries don't change.

### CODE REVIEW

**Comment by alejandrohdezma** [comment]
Yeah, that's my intention. I'm working on it in the `47degrees/fetch` repository and will then propagate to the rest.

File: build.sbt [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
-  .in(file(".docs"))
-  .aggregate(testkit)
-  .settings(moduleName := "docker-testkit-munit-project-docs")
-  .settings(mdocIn := file(".docs"))
+name := "docker-testkit-munit"
+
+testFrameworks += new TestFramework("munit.Framework")
+fork in Test := true
+
+val munit   = "org.scalameta" %% "munit"               % "[0.2.0,)" % Provided // scala-steward:off
```
Comment (BenFradet) [inline build.sbt:13]: how did you figure out this version? :thinking:

File: project/plugins.sbt [hunk 2]
```diff
@@ -1,9 +1,8 @@
 addSbtPlugin("com.geirsson"              % "sbt-ci-release"    % "1.5.3")
-addSbtPlugin("org.scalameta"             % "sbt-mdoc"          % "2.1.5")
-addSbtPlugin("org.scalameta"             % "sbt-scalafmt"      % "2.3.4")
+addSbtPlugin("org.scalameta"             % "sbt-mdoc"          % "2.2.0")
```
Comment (BenFradet) [inline project/plugins.sbt:2]: weird that scala-steward didn't care for those

File: build.sbt (hunk 1 above)
Comment (alejandrohdezma) [reply build.sbt:13]: I changed the version to the first one being published for Scala `2.13` and run `ci-test` using that version 😸

File: project/plugins.sbt (hunk 2 above)
Comment (alejandrohdezma) [reply project/plugins.sbt:2]: I think Scala Steward isn't working as it should... hopefully that will change with the next version of `.github`... 😈

**Comment by alejandrohdezma** [comment]
@BenFradet is this good to merge?

**Pull request merged**